src/MSALWrapper/TokenFetcher.cs

```diff
@@ -105,8 +105,8 @@ public record Result
                 try
                 {
                     // GetTokenAsync returns an empty list instead of null so no null check required here.
-                    results.AddRange(await executor.GetTokenAsync());
-                }
+                    results.AddRange(executor.GetTokenAsync().Result);

```

Comment:
@goagain @reillysiemens I've updated the description of this PR, see if that makes more sense. Tl;Dr - the same thread that acquires the mutex must also release the mutex and awaiting, since our library code uses `ConfigureAwait(false)` downstream, breaks this constraint.